odl/solvers/util/callback.py

```diff
@@ -438,19 +440,137 @@ def reset(self):
         """Set `iter` to 0 and create a new figure."""
         self.iter = 0
         self.fig = None
+        self.space_of_last_x = None
 
     def __repr__(self):
         """Return ``repr(self)``."""
-
         return '{}(display_step={}, saveto={}, fig={}, *{!r}, **{!r})'.format(
             self.__class__.__name__,
-            self.saveto,
             self.display_step,
+            self.saveto,
             self.fig,
             self.args,
             self.kwargs)
 
 
+class CallbackSaveToDisk(SolverCallback):
+
+    """Save the iterates to disk."""
+
+    def __init__(self, saveto, save_step=1, impl='pickle', **kwargs):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        saveto : string
+            Path to a directory where the figures are to be saved, followed by
+            a file name and potentially a file format. Standard python string
+            formatting is possible to use in order to index the iterates with
+            iteration number.
```

Comment:
Imo the format option is "advanced", so I'll keep it out for now. The other suggestion is great however.